numpyro/contrib/hsgp/approximation.py

```diff
@@ -0,0 +1,179 @@
+    squared exponential kernel and the spectrum of the Drichlet Laplacian operator to
+    obtain a low-rank approximation of the Gram matrix. For more details see [1, 2].
+
+    **References:**
+
+    1. Solin, A., Särkkä, S. Hilbert space methods for reduced-rank Gaussian process regression.
+    Stat Comput 30, 419-446 (2020).
+
+    2. Riutort-Mayol, G., Bürkner, PC., Andersen, M.R. et al. Practical Hilbert space
+    approximate Bayesian Gaussian processes for probabilistic programming. Stat Comput 33, 17 (2023).
+
+    :param ArrayImpl x: input data
+    :param float alpha: amplitude of the squared exponential kernel
+    :param float length: length scale of the squared exponential kernel
+    :param float ell: The length of the interval divided by 2
+    :param int m: The number of eigenvalues to compute
+    :param bool non_centered: Whether to use a non-centered parameterization. By default, it is set to True
+    :return: The low-rank approximation linear model
+    :rtype: ArrayImpl
+    """
+    phi = eigenfunctions(x=x, ell=ell, m=m)
+    spd = jnp.sqrt(
+        diag_spectral_density_squared_exponential(
+            alpha=alpha, length=length, ell=ell, m=m
+        )
+    )
+    return linear_approximation(phi=phi, spd=spd, m=m, non_centered=non_centered)
+
+
+def hsgp_approximation_matern(
+    x: ArrayImpl,
+    nu: float,
+    alpha: float,
+    length: float,
+    ell: float,
+    m: int,
+    non_centered: bool = True,
+):
+    """
+    Hilbert space Gaussian process approximation using the Matérn kernel.
+
+    The main idea of the approach is to combine the associated spectral density of the
+    Matérn kernel kernel and the spectrum of the Drichlet Laplacian operator to obtain
+    a low-rank approximation of the Gram matrix. For more details see [1, 2].
+
+    **References:**
+
+    1. Solin, A., Särkkä, S. Hilbert space methods for reduced-rank Gaussian process regression.
+    Stat Comput 30, 419-446 (2020).
+
+    2. Riutort-Mayol, G., Bürkner, PC., Andersen, M.R. et al. Practical Hilbert space
+    approximate Bayesian Gaussian processes for probabilistic programming. Stat Comput 33, 17 (2023).
+
+    :param ArrayImpl x: input data
+    :param float nu: smoothness parameter
+    :param float alpha: amplitude of the squared exponential kernel
+    :param float length: length scale of the squared exponential kernel
+    :param float ell: The length of the interval divided by 2
+    :param int m: The number of eigenvalues to compute
+    :param bool non_centered: Whether to use a non-centered parameterization. By default, it is set to True.
+    :return: The low-rank approximation linear model
+    :rtype: ArrayImpl
+    """
+    phi = eigenfunctions(x=x, ell=ell, m=m)
+    spd = jnp.sqrt(
+        diag_spectral_density_matern(nu=nu, alpha=alpha, length=length, ell=ell, m=m)
+    )
+    return linear_approximation(phi=phi, spd=spd, m=m, non_centered=non_centered)
+
+
+def hsgp_approximation_periodic_non_centered(
```

Comment:
lots of these method names seem pretty long... do you really need "approximation" everywhere?

Reply:
Ups! You are right 🙈. I can remove the “approximation” from the function names.

Reply:
Simplified main function names in https://github.com/pyro-ppl/numpyro/pull/1794/commits/2b1e37d3b34c5d92797ce7f3b08480035cad98ae